src/main/java/com/github/speedwing/log4j/cloudwatch/appender/CloudwatchAppender.java

```diff
@@ -85,54 +90,60 @@ public void setMessagesBatchSize(int messagesBatchSize) {
 
     @Override
     protected void append(LoggingEvent event) {
+        // Make sure that we don't get stuck in an infinite loop
+        if (event.getLocationInformation().getClassName().startsWith("com.amazonaws")) {
+            return;
+        }
+    
         if (cloudwatchAppenderInitialised.get()) {
```

Comment:
I would add a check for `&& !this.closed` so that if it's closed no more messages will be appended and should prevent the loop/dead lock you guys mentioned